spec/git-repository-provider-spec.coffee

```diff
@@ -54,3 +54,35 @@ describe "GitRepositoryProvider", ->
           directory = new Directory dirPath
           provider.repositoryForDirectory(directory).then (result) ->
             expect(result).toBe null
+
+    describe "when specified a Directory without existsSync()", ->
+      it "returns null", ->
+        provider = new GitRepositoryProvider atom.project
+
+        # Tolerate an implementation of Directory that does not implement existsSync().
```

Comment:
Could you put the code that sets up `directory` into a `beforeEach`. My criteria for extracting something like that is roughly:
* It's the same for all of the tests in the `describe` block.
* It corresponds directly to the description of the `describe` block.